lib/snprintf.c

```diff
@@ -1097,6 +1097,9 @@ fmtflt(char *str, size_t *len, size_t size, LDOUBLE fvalue, int width,
 		infnan = (flags & PRINT_F_UP) ? "INF" : "inf";
 
 	if (infnan != NULL) {
+		int i = MAX_CONVERT_LENGTH - 1;
+		while (i >= 0) /* Initialize buffer, do not use memset (compatility) */
+			iconvert[i--] = '\0';
```

Comment:
Why not use memset?

Also, why not do this at the beginning at line 1084?

Should the same be done for fconvert, at least for consistency?

Reply:
Also (not a requirement, just a request), could you send the patch back to the original author as well?

http://www.jhweiss.de/software/snprintf.html

Reply:
Function memset() is not used anywhere in libcompat. So I figured maybe someone had already come to the conclusion that memset() is not available at every C installation or other compatibility issue. A little backwards logic here, I know. Just being overly careful.

However, memset() is used in libcheck in three places. If you say that it is okay to use memset(), I'm all for it.

I will move it to the beginning of the function, line 1084. A good idea, since the buffer is also used later in the function.

Same for fconvert? I will change that too.